CMakeLists.txt

```diff
@@ -232,28 +245,40 @@ if (MINGW)
     add_library(${PROJECT} ${OLOCO_SOURCES})
     target_link_libraries(${PROJECT} winmm)
 else ()
-    add_executable(${PROJECT} ${OLOCO_SOURCES} ${OLOCO_MM_SOURCES} ${LOCO_SECTIONS})
-    set_target_properties(${PROJECT} PROPERTIES LINK_FLAGS ${LOCO_SEGMENT_LINKER_FLAGS})
+    if (WIN32)
+        add_library(${PROJECT} SHARED ${OLOCO_SOURCES})
+    else ()
+        add_executable(${PROJECT} ${OLOCO_SOURCES} ${OLOCO_MM_SOURCES} ${LOCO_SECTIONS})
+        set_target_properties(${PROJECT} PROPERTIES LINK_FLAGS ${LOCO_SEGMENT_LINKER_FLAGS})
+    endif ()
 endif ()
 
 target_link_libraries(${PROJECT} ${SDL2_LIBRARIES} ${SDL2_MIXER_LIBRARIES})
-target_link_libraries(${PROJECT} yaml-cpp ${YAML_CPP_LIBRARIES})
+target_link_libraries(${PROJECT} ${YAML_CPP_LIBRARIES})
 target_link_libraries(${PROJECT} ${PNG_LIBRARIES})
 
+if (WIN32)
+    target_link_libraries(${PROJECT} ${BREAKPAD_LIBRARIES})
+    target_link_libraries(${PROJECT} winmm SDL2::SDL2main)
+endif ()
 
-if (NOT MINGW)
+if (NOT WIN32)
     add_dependencies(${PROJECT} segfiles)
 endif()
 
-if (NOT APPLE)
+if (NOT APPLE AND NOT MSVC)
     target_link_libraries(${PROJECT} stdc++fs)
 endif()
 
 if (MINGW)
     target_compile_definitions(${PROJECT} PRIVATE _WIN32_WINNT=0x0501 SDL_MAIN_HANDLED=1)
     target_link_libraries(${PROJECT} imm32 winmm version)
 else ()
-    target_compile_definitions(${PROJECT} PRIVATE _NO_LOCO_WIN32_=1)
+    if (MSVC)
+        target_compile_definitions(${PROJECT} PRIVATE _WINDLL _MBCS DEBUG_WIN32=1 _NO_LOCO_WIN32_=1 _CRT_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_DEPRECATE _USE_MATH_DEFINES SDL_MAIN_HANDLED _WINSOCK_DEPRECATED_NO_WARNINGS USE_BREAKPAD)
```

Comment:
The NO_LOCO_WIN32 should be on for none MSVC and off for MSVC

Reply:
After making this change the compiled binary was able to be run.

Reply:
Will fix